q2_fmt/tests/test_engraftment.py

```diff
@@ -1023,3 +1023,81 @@ def test_drop_incomplete_timepoints_list(self):
                       float("Nan")]}).set_index('id')
         metadata_df = _drop_incomplete_timepoints(metadata_df, "group", [3, 2])
         self.assertEqual(metadata_df["group"].dropna().unique(), [float(1)])
+
+    def test_rename_features_with_delim(self):
+        metadata_df = pd.DataFrame({
+                'id': ['sample1', 'sample2', 'sample3',
+                       'donor1'],
+                'Ref': ['donor1', 'donor1', 'donor1', float("Nan")],
+                'subject': ['sub1', 'sub1', 'sub1', float("Nan")],
+                'group': [1, 1, 1, float("Nan")]}).set_index('id')
+        metadata = Metadata(metadata_df)
+        table_df = pd.DataFrame({
+                'id': ['sample1', 'sample2', 'sample3',
+                       'donor1'],
+                'Feature;1': [0, 0, 1, 1],
+                'Feature;2': [0, 1, 1, 1],
+                'Feature;3': [0, 0, 1, 0]}).set_index('id')
+        feature_peds_df = feature_peds(table=table_df, metadata=metadata,
+                                       time_column="group",
+                                       reference_column="Ref",
+                                       subject_column="subject")
+        _rename_features(data=feature_peds_df, level_delimiter=";")
```

Comment:
It might be good to include tests for `peds_heatmap` since it seems like that's where this helper is actually being used in the context of your plugin. That being said, I know it can be difficult to test visualizers - I wonder if it's possible to test the contents of the raw JSON for this... that might be doable.

Reply:
I am open to this but every time this has been brought up we have decided its not worth the squeeze. Let me know.

Reply:
Let me think about this and get back to you - otherwise this is a reasonable way to test the helper method in a different context!

Reply:
okay i looked into this a little bit and i don't think the juice is worth the squeeze - so these tests should be sufficient!